vyper/exceptions.py

```diff
@@ -51,74 +51,111 @@ def __str__(self):
 
 class PythonSyntaxException(ParserException):
     """
-    Conversion from error using ast.parse()
+    Invalid syntax.
+
+    This error is converted from errors raised during ast.parse
     """
     def __init__(self, syntax_error: SyntaxError, source_code: str):
         item = types.SimpleNamespace()  # TODO: Create an actual object for this
         item.lineno = syntax_error.lineno
         item.col_offset = syntax_error.offset
-        item.source_code = source_code
+        item.full_source_code = source_code
         super().__init__(message=f'SyntaxError: {syntax_error.msg}', item=item)
 
 
-class VariableDeclarationException(ParserException):
-    pass
+class SyntaxException(ParserException):
+    """Valid Python syntax, but invalid Vyper syntax."""
 
 
 class StructureException(ParserException):
-    pass
-
-
-class ConstancyViolationException(ParserException):
-    pass
-
-
-class NonPayableViolationException(ParserException):
-    pass
+    """TODO"""
+    # unknown / duplicate interface
+    # invalid map declaration
+    # invalid decorator
+    # use of semi-colon
+    # raise without reason
+    # iteration over nested list
+    # tuple length mismatch
+    # invalid assignment with/without value
+    # improperly formatted struct declaration
+    # invalid global keyword
+    # empty list/tuple
+    # unsupported statement type
+    # altering list during iteration
+    # invalid use of range()
 
 
 class InvalidLiteralException(ParserException):
-    pass
+    """Invalid literal value."""
+    # number out of range
+    # too many decimals
+    # invalid hexstring length
+    # comparison against None
 
 
 class InvalidTypeException(ParserException):
-    pass
+    """Invalid type declaration."""
 
 
-class TypeMismatchException(ParserException):
-    pass
+class VariableDeclarationException(ParserException):
+    """Invalid variable declaration."""
+    # invalid name
+    # already declared / shadows existing value
+    # undeclared
 
 
 class FunctionDeclarationException(ParserException):
-    pass
+    """Invalid function declaration."""
+    # invalid arguments
+    # already declared / shadows existing value
 
 
 class EventDeclarationException(ParserException):
-    pass
+    """Invalid event declaration."""
+    # undeclared
+    # already declared / shadows existing value
+    # invalid argument types
+    # invalid indexed arg
 
 
-class VersionException(ParserException):
-    pass
+class TypeMismatchException(ParserException):
+    """TODO"""
+    # different types for operation
+    # same type but different units
+    # unexpected return value, expected return value
+    # left / right length mismatch
+    # list items mismatched
+    # type cannot be packed
 
 
-class SyntaxException(ParserException):
-    pass
+class ConstancyViolationException(ParserException):
+    """State-changing action inside a constant function."""
+
+
+class NonPayableViolationException(ParserException):
```

Comment:
`NonPayableViolationException` is only raised in one place in the entire codebase.

Reply:
Seems a little non-descript to me